File: leetcode/2024 Maximize the Confusion of an Exam/solution.java
class Solution {
    public int maxConsecutiveAnswers(String answerKey, int k) { // 滑动窗口虫取法！！！
        int[] count = new int[2];
        int maxCount = 0;
        int left = 0, right = 0, index = 0;
        while (right < answerKey.length()) {
            index = (answerKey.charAt(right) == 'T') ? 0 : 1;
            count[index]++;
            maxCount = Math.max(maxCount, count[index]);
            while (right-left+1 - maxCount > k) {
                index = (answerKey.charAt(left) == 'T') ? 0 : 1;
                count[index]--;
                left++;
            }
            right++;
        }
        return right-left;
    }
}

File: leetcode/2024 Maximize the Confusion of an Exam/solution.go
func maxConsecutiveAnswers(answerKey string, k int) int { // 滑动窗口虫取法！！！
    count := [2]int{}
    maxCount, index, left, right := 0, 0, 0, 0
    for right < len(answerKey) {
        if (answerKey[right] == 'T') {
            index = 0
        } else {
            index = 1
        }
        count[index]++
        if count[index] > maxCount {
            maxCount = count[index];
        }
        if right-left+1 - maxCount > k {
            if (answerKey[left] == 'T') {
                index = 0
            } else {
                index = 1
            }
            count[index]--
            left++
        }
        right++
    }
    return right-left
}